src/sagemaker/serve/builder/jumpstart_builder.py

```diff
@@ -234,6 +254,213 @@ def _build_for_tgi_jumpstart(self):
+                )
+                break
+            except LocalModelOutOfMemoryException as e:
+                logger.warning(
+                    "Deployment unsuccessful with %s: %s. SAGEMAKER_MODEL_SERVER_WORKERS: %s. "
+                    "Out of memory when loading the model: %s",
+                    num_shard_env_var_name,
+                    tensor_parallel_degree,
+                    sagemaker_model_server_workers,
+                    str(e),
+                )
+                break
+            except LocalModelInvocationException as e:
+                logger.warning(
+                    "Deployment unsuccessful with %s: %s. SAGEMAKER_MODEL_SERVER_WORKERS: %s. "
+                    "Failed to invoke the model server: %s"
+                    "Please check that model server configurations are as expected "
+                    "(Ex. serialization, deserialization, content_type, accept).",
+                    num_shard_env_var_name,
+                    tensor_parallel_degree,
+                    sagemaker_model_server_workers,
+                    str(e),
+                )
+                break
+            except LocalModelLoadException as e:
+                logger.warning(
+                    "Deployment unsuccessful with %s: %s. SAGEMAKER_MODEL_SERVER_WORKERS: %s. "
+                    "Failed to load the model: %s.",
+                    num_shard_env_var_name,
+                    tensor_parallel_degree,
+                    sagemaker_model_server_workers,
+                    str(e),
+                )
+                break
+            except SkipTuningComboException as e:
+                logger.warning(
+                    "Deployment with %s: %s. SAGEMAKER_MODEL_SERVER_WORKERS: %s. "
+                    "was expected to be successful. However failed with: %s. "
+                    "Trying next combination.",
+                    num_shard_env_var_name,
+                    tensor_parallel_degree,
+                    sagemaker_model_server_workers,
+                    str(e),
+                )
+                break
+            except Exception:  # pylint: disable=W0703
+                logger.exception(
+                    "Deployment unsuccessful with %s: %s. SAGEMAKER_MODEL_SERVER_WORKERS: %s. "
+                    "with uncovered exception",
+                    num_shard_env_var_name,
+                    tensor_parallel_degree,
+                    sagemaker_model_server_workers,
+                )
+                break
+
+        if best_tuned_combination:
+            self.pysdk_model.env.update(
+                {
+                    num_shard_env_var_name: str(best_tuned_combination[1]),
+                    "SAGEMAKER_MODEL_SERVER_WORKERS": str(best_tuned_combination[2]),
+                }
+            )
+
+            _pretty_print_benchmark_results(
+                benchmark_results, [num_shard_env_var_name, "SAGEMAKER_MODEL_SERVER_WORKERS"]
+            )
+            logger.info(
```

Comment:
Minor if we are already pretty printing results, why do we need to log similar info again?

Reply:
`Pretty print` prints the entire benchmark results, the log prints only the best performing benchmark result. Similar to existing logic https://github.com/aws/sagemaker-python-sdk/blob/ca9ae23e58071d67bb8b33cae8c2256f095b76ee/src/sagemaker/serve/builder/djl_builder.py#L453-L463